common/scala/src/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -42,55 +42,60 @@ class KafkaConsumerConnector(
     with Logging
     with TransactionCounter {
 
-    override def onMessage(process: Message => Unit) = {
+    def getMessages(duration: Duration = 100 milliseconds) = consumer.poll(duration.toMillis)
+
+    override def onMessage(process: Array[Byte] => Boolean) = {
         val self = this
         val thread = new Thread() {
             override def run() = {
-                for (metadata <- stream) {
-                    val raw = new String(metadata.message, "utf-8")
-                    val msg = Message(raw)
-                    msg match {
-                        case Success(m) =>
-                            info(self, s"received /${metadata.topic}[${metadata.partition}][${metadata.offset}]: ${m.path}")(m.transid)
-                            process(m)
-                        case Failure(t) =>
-                            info(this, errorMsg(raw, t))
+                while (!disconnect) {
```

Comment:
I'm wondering if the CPU usage of the new loop here is the same as the old loop.  ?

That would be possible if the old loop used a blocking interface to fetch messages from the stream, but the new implementation of getMessages() is non-blocking.      Not sure if this is the case -- but would like to understand.